drivers/xen/gntdev-dmabuf.c

```diff
@@ -432,6 +524,120 @@ dmabuf_exp_alloc_backing_storage(struct gntdev_priv *priv, int dmabuf_flags,
 	return map;
 }
 
+static int dmabuf_release_notifier_cb(struct notifier_block *nfb,
+		unsigned long val, void *gnt_buf);
+
+static struct notifier_block notify_release_nfb = {
+	.notifier_call = dmabuf_release_notifier_cb,
+};
+
+static void dmabuf_map_remove(struct gntdev_priv *priv,
+		  struct gntdev_dmabuf *gntdev_dmabuf)
+{
+	dmabuf_exp_remove_map(priv, gntdev_dmabuf->u.map.map);
+	dmabuf_map_free_gntdev_dmabuf(gntdev_dmabuf);
+}
+
+static struct gntdev_dmabuf *
+dmabuf_alloc_gntdev_from_buf(struct gntdev_priv *priv, int fd, int dmabuf_flags,
+						 int count, unsigned int data_ofs)
+{
+	struct gntdev_dmabuf *gntdev_dmabuf;
+	struct dma_buf_attachment *attach;
+	struct dma_buf *dma_buf;
+	struct sg_table *sgt;
+	int ret = 0;
+
+	gntdev_dmabuf = kzalloc(sizeof(*gntdev_dmabuf), GFP_KERNEL);
+	if (unlikely(!gntdev_dmabuf))
```

Comment:
Do you really need to use `unlikely` in this function? Is it on hot path?

Reply:
According to what I've read about unlikely and branch prediction mechanisms in modern CPU - unlikely works only on the first time and doesn't affect performance in loops. Probably I'm wrong.